crates/oxc_linter/src/rules/jest/prefer_expect_resolves.rs

```diff
@@ -0,0 +1,267 @@
+    /// 2. `await` the promise and assert against its result
+    /// (`expect(await ...).<matcher>` style)
+    ///
+    /// While the second style is arguably less dependent on `jest`, if the promise
+    /// rejects it will be treated as a general error, resulting in less predictable
+    /// behaviour and output from `jest`.
+    ///
+    /// Additionally, favoring the first style ensures consistency with its `rejects`
+    /// counterpart, as there is no way of "awaiting" a rejection.
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// // valid
+    /// it('passes', async () => {
+    ///     await expect(someValue()).resolves.toBe(true);
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///
+    ///     await expect(myPromise).resolves.toBe(true);
+    /// });
+    ///
+    /// it('errors', async () => {
+    ///     await expect(Promise.reject(new Error('oh noes!'))).rejects.toThrowError(
+    ///         'oh noes!',
+    ///     );
+    /// });
+    ///
+    /// // invalid
+    /// it('passes', async () => {
+    ///     expect(await someValue()).toBe(true);
+    /// });
+    /// it('is true', async () => {
+    ///     const myPromise = Promise.resolve(true);
+    ///     expect(await myPromise).toBe(true);
+    /// });
+    /// ```
+    PreferExpectResolves,
+    style,
+);
+
+impl Rule for PreferExpectResolves {
+    fn run_once(&self, ctx: &LintContext) {
+        for possible_jest_node in &collect_possible_jest_call_node(ctx) {
+            Self::run(possible_jest_node, ctx);
+        }
+    }
+}
+
+impl PreferExpectResolves {
+    fn run<'a>(possible_jest_node: &PossibleJestNode<'a, '_>, ctx: &LintContext<'a>) {
+        let node = possible_jest_node.node;
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        let Some(expect_fn_call) = parse_expect_jest_fn_call(call_expr, possible_jest_node, ctx)
+        else {
+            return;
+        };
+        let Some(parent) = expect_fn_call.head.parent else {
+            return;
+        };
+
+        if !parent.is_call_expression() {
+            return;
+        }
+
+        let Some(Expression::CallExpression(call_expr)) = expect_fn_call.head.parent else {
+            return;
+        };
+        let Some(argument) = call_expr.arguments.first() else {
+            return;
+        };
+        let Argument::Expression(Expression::AwaitExpression(await_expr)) = argument else {
+            return;
+        };
+
+        ctx.diagnostic_with_fix(ExpectResolves(await_expr.span), || {
+            let content = Self::build_code(call_expr, ctx);
+            Fix::new(content, call_expr.span)
+        });
+    }
+
+    fn build_code<'a>(call_expr: &CallExpression<'a>, ctx: &LintContext<'a>) -> String {
+        let mut formatter = ctx.codegen();
+        let ident = call_expr.callee.get_identifier_reference().unwrap();
```

Comment:
Feels like using `unwrap` not 100% safety, even I can't show a case now.

Reply:
I opened a new [PR](https://github.com/oxc-project/oxc/pull/2715) for this, which I can get the identifier directly, and it has been parsed by `parse_expect_jest_fn`.

Reply:
merged.